watertap/unit_models/dewatering.py

```diff
@@ -0,0 +1,187 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Dewatering unit model for BSM2. Based on IDAES separator unit
+
+Model based on 
+
+J. Alex, L. Benedetti, J.B. Copp, K.V. Gernaey, U. Jeppsson,
+I. Nopens, M.N. Pons, C. Rosen, J.P. Steyer and
+P. A. Vanrolleghem
+Benchmark Simulation Model no. 2 (BSM2)
+"""
+
+from enum import Enum
+from pandas import DataFrame
+
+# Import IDAES cores
+from idaes.core import (
+    declare_process_block_class,
+    MaterialBalanceType,
+)
+from idaes.models.unit_models.separator import SeparatorData
+
+from idaes.core.util.tables import create_stream_table_dataframe
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.solvers import get_solver
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslog
+
+from pyomo.environ import (
+    Constraint,
+    Param,
+    Block,
+    value,
+    units as pyunits,
+    check_optimal_termination,
+    Set,
+)
+
+from idaes.core.util.exceptions import (
+    ConfigurationError,
+    PropertyNotSupportedError,
+    InitializationError,
+)
+
+__author__ = "Alejandro Garciadiego"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+# Enumerate options for balances
+class SplittingType(Enum):
+    """
+    Enum of supported material split types.
+    """
+
+    totalFlow = 1
+    phaseFlow = 2
+    componentFlow = 3
+    phaseComponentFlow = 4
+
+
+@declare_process_block_class("Dewatering_Unit")
+class DewateringUnit(SeparatorData):
+    """
+    Dewatering unit block for BSM2
+    """
+
+    def build(self):
+        """
+        Begin building model.
+        Args:
+            None
+        Returns:
+            None
+        """
+
+        # Call UnitModel.build to setup dynamics
+        super(DewateringUnit, self).build()
+
+        if "underflow" and "overflow" not in self.config.outlet_list:
+            raise ConfigurationError(
+                "{} encountered unrecognised "
+                "outlet_list. This should not "
+                "occur - please use overflow "
+                "and underflow as outlets.".format(self.name)
+            )
```

Comment:
I suppose we could keep this if we incorporate my change above, in case an advanced user knew they could modify the config's outlet list to whatever they want. This would catch that case.